### PR TITLE
Charge for hashing init code in CREATE2

### PR DESCRIPTION
**Pull request by gumb0**
Changing the gas cost of CREATE2 according to the latest changes in spec https://github.com/ethereum/EIPs/pull/1375

cc @winsvega

### CODE REVIEW

**Comment by chfast** [comment]
It cannot find sha3WordGas symbol :/

**Comment by gumb0** [comment]
Linker can't find the symbol, but it's `constexpr`, weird. And it's fine with `VMSchedule::createGas` in the same function 😕

File: libaleth-interpreter/VMCalls.cpp [hunk 1]
```diff
@@ -116,7 +116,11 @@ void VM::caseCreate()
 
     u256 salt;
     if (m_OP == Instruction::CREATE2)
+    {
         salt = m_SP[3];
+        // charge for hashing initCode = GSHA3WORD * ceil(len(init_code) / 32)
+        m_runGas += toInt63((static_cast<u512>(initSize) + 31) / 32 * uint64_t(VMSchedule::sha3WordGas));
```
Comment (chfast) [inline libaleth-interpreter/VMCalls.cpp]: Write it as `u512(initSize)` because there is not data loss here.

**Comment by gumb0** [comment]
So `uint64_t{VMSchedule::sha3WordGas}` fixes the linker error.

Another option is to add definitions of `VMSchedule` static members to cpp file like
```
constexpr int64_t VMSchedule::sha3WordGas;
...
```

**Comment by chfast** [comment]
I don't care which workaround is used, but I've found this for `SHA3` instructions:
```
        CASE(SHA3)
        {
            ON_OP();
            constexpr int64_t sha3Gas = VMSchedule::sha3Gas;
            constexpr int64_t sha3WordGas = VMSchedule::sha3WordGas;
            m_runGas = toInt63(sha3Gas + (u512(m_SP[1]) + 31) / 32 * sha3WordGas);
```

**Comment by gumb0** [comment]
I see, this is third kind of workaround.

So as I understand `constexpr int64_t sha3WordGas = VMSchedule::sha3WordGas;` works and `(u512(m_SP[1]) + 31) / 32 * VMSchedule::sha3WordGas` doesn't, because the second one probably tries to pass `VMSchedule::sha3WordGas` by reference to `boost::multiprecision::operator*` and creating a reference requires definition outside of class in C++11

**Comment by chfast** [comment]
I don't know why it work and I don't want to know. I think it was me who add this but I have not investigated the reasons.

File: libevm/LegacyVMCalls.cpp [hunk 2]
```diff
@@ -127,7 +127,11 @@ void LegacyVM::caseCreate()
 
     u256 salt;
     if (m_OP == Instruction::CREATE2)
+    {
         salt = m_SP[3];
+        // charge for hashing initCode = GSHA3WORD * ceil(len(init_code) / 32)
+        m_runGas += toInt63((static_cast<u512>(initSize) + 31) / 32 * m_schedule->sha3WordGas);
```
Comment (chfast) [inline libevm/LegacyVMCalls.cpp]: Same `static_cast` -> `u512{}` change here.

Comment (gumb0) [reply libevm/LegacyVMCalls.cpp]: done

**Pull request merged**